src/game/client/components/scoreboard.cpp

```diff
@@ -342,80 +469,131 @@ void CScoreboard::OnRender()
 	if(!Active())
 		return;
 
+	// don't render scoreboard if menu is open
+	if(m_pClient->m_pMenus->IsActive())
+		return;
+
 	// if the score board is active, then we should clear the motd message aswell
 	if(m_pClient->m_pMotd->IsActive())
 		m_pClient->m_pMotd->Clear();
 
 
-	float Width = 400*3.0f*Graphics()->ScreenAspect();
-	float Height = 400*3.0f;
+	CUIRect Screen = *UI()->Screen();
+	Graphics()->MapScreen(Screen.x, Screen.y, Screen.w, Screen.h);
 
-	Graphics()->MapScreen(0, 0, Width, Height);
-
-	float w = 700.0f;
+	float Width = Screen.w;
+	float y = 85.f;
+	float w = 364.0f;
+	float FontSize = 86.0f;
 
 	if(m_pClient->m_Snap.m_pGameData)
 	{
 		if(!(m_pClient->m_GameInfo.m_GameFlags&GAMEFLAG_TEAMS))
-			RenderScoreboard(Width/2-w/2, 150.0f, w, 0, 0);
+		{
+			float ScoreboardHeight = RenderScoreboard(Width/2-w/2, y, w, 0, 0, -1);
+
+			float SpectatorHeight = RenderSpectators(Width/2-w/2, y+3.0f+ScoreboardHeight, w);
+			RenderGoals(Width/2-w/2, y+3.0f+ScoreboardHeight, w);
+
+			// scoreboard size
+			m_TotalRect.x = Width/2-w/2;
+			m_TotalRect.y = y;
+			m_TotalRect.w = w;
+			m_TotalRect.h = ScoreboardHeight+SpectatorHeight+3.0f;
+		}
 		else if(m_pClient->m_Snap.m_pGameDataTeam)
 		{
 			const char *pRedClanName = GetClanName(TEAM_RED);
```

Comment:
These lines are there twice, move it outside the if clauses. And check if the clan name is empty there and assign "Red/Blue team".

Reply:
> And check if the clan name is empty there and assign "Red/Blue team".

I don't think that's a good idea because we need to keep the `str_format` for later, for the `Localize`:

```
				if(pRedClanName)
					str_format(aText, sizeof(aText), Localize("%s wins!"), pRedClanName);
```

Reply:
There currently is:
-Localize("%s wins!")
-Localize("Red team") <- pRedTeamName
-Localize("Red team wins!") <- this feels redundant
  = str_format(aBuf, sizeof(aBuf), Localize("%s wins!"), pRedTeamName);

Reply:
You're right.